src/app/app.datatypes.ts

```diff
@@ -63,14 +64,32 @@ export class GetAddressResponseTransaction {
   txid: string;
 }
 
-export function parseGetAddressTransaction(raw: GetAddressResponseTransaction): Transaction {
+export function parseGetAddressTransaction(raw: GetAddressResponseTransaction, address: string): Transaction {
+
+  //Detect if the address sent or received the coins.
+  let incoming: boolean = true;
+  if (raw.inputs[0].owner.toLowerCase() == address.toLowerCase()) { incoming = false; }
+
+  //Calculate the amount of incoming or outgoing coins, as appropriate.
+  let coins:number = 0;
+  for (let output of raw.outputs) {
+    if ((incoming == true) && (output.dst.toLowerCase() == address.toLowerCase())) {
```

Comment:
* Don't need to compare with true as incoming is already a boolean
* Brackets are redundant